### ISSUE TITLE
Provider BinDeps.BuildProcess failed to satisfy dependency ddApp

### ISSUE THREAD

**Issue by blegat** [rdeits/DrakeVisualizer.jl#7]
I got this error when running `Pkg.build("DrakeVisualizer")`

```
INFO: Building PyCall
Fetching package metadata .......
Solving package specifications: ..........

# All requested packages already installed.
# packages in environment at /home/blegat/.julia/v0.5/Conda/deps/usr:
#
numpy                     1.11.2                   py27_0  
INFO: PyCall is using /home/blegat/.julia/v0.5/Conda/deps/usr/bin/python (Python 2.7.11) at /home/blegat/.julia/v0.5/Conda/deps/usr/bin/python, libpython = /home/blegat/.julia/v0.5/Conda/deps/usr/lib/libpython2.7
INFO: Building PyLCM
WARNING: `@linux` is deprecated, use `@static is_linux()` instead
 in depwarn(::String, ::Symbol) at ./deprecated.jl:64
 in @linux(::Any, ::Any) at ./deprecated.jl:484
 in include_from_node1(::String) at ./loading.jl:488
 in evalfile(::String, ::Array{String,1}) at ./loading.jl:504 (repeats 2 times)
 in cd(::##2#4, ::String) at ./file.jl:59
 in (::##1#3)(::IOStream) at ./none:13
 in open(::##1#3, ::String, ::String) at ./iostream.jl:113
 in eval(::Module, ::Any) at ./boot.jl:234
 in process_options(::Base.JLOptions) at ./client.jl:239
 in _start() at ./client.jl:318
while loading /home/blegat/.julia/v0.5/PyLCM/deps/build.jl, in expression starting on line 17
-I/usr/include/python2.7 -I/usr/include/x86_64-linux-gnu/python2.7
-I/usr/include/python2.7 -I/usr/include/x86_64-linux-gnu/python2.7
-I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include
-I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include
INFO: Building DrakeVisualizer
INFO: Attempting to Create directory /home/blegat/.julia/v0.5/DrakeVisualizer/deps/downloads
INFO: Downloading file http://people.csail.mit.edu/patmarion/software/director/releases/director-0.1.0-linux.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9094k  100 9094k    0     0  1557k      0  0:00:05  0:00:05 --:--:-- 1518k
INFO: Done downloading file http://people.csail.mit.edu/patmarion/software/director/releases/director-0.1.0-linux.tar.gz
INFO: Attempting to Create directory /home/blegat/.julia/v0.5/DrakeVisualizer/deps/usr
==========================================================[ ERROR: DrakeVisualizer ]===========================================================

LoadError: Provider BinDeps.BuildProcess failed to satisfy dependency ddApp
while loading /home/blegat/.julia/v0.5/DrakeVisualizer/deps/build.jl, in expression starting on line 46

===============================================================================================================================================

===============================================================[ BUILD ERRORS ]================================================================

WARNING: DrakeVisualizer had build errors.

 - packages with build errors remain installed in /home/blegat/.julia/v0.5
 - build the package(s) and all dependencies with `Pkg.build("DrakeVisualizer")`
 - build a single package by running its `deps/build.jl` script

===============================================================================================================================================
```

I have written a little C file to see what is the error returned by dlopen and I get

```
could not open file [libddApp.so]: libQVTK.so.5.8: cannot open shared object file: No such file or directory
```

As it was suggested in the README, I had done

```
sudo apt-get install libvtk5-qt4-dev python-vtk
```

but it had installed

```
$ ldconfig -p | grep libQVTK                                               
    libQVTK.so.5.10 (libc6,x86-64) => /usr/lib/libQVTK.so.5.10
    libQVTK.so (libc6,x86-64) => /usr/lib/libQVTK.so
```

so I guess that the precompiled binaries are pointing to the wrong version of `libQVTK` :(

**Comment by blegat** [comment]
That is weird because travis is working on Linux

**Comment by blegat** [comment]
```
julia> BinDeps.debug("DrakeVisualizer")
INFO: Reading build script...
The package declares 3 dependencies.
 - Library "python"
    - Satisfied by:
      - BinDeps.AptGet package python2.7 at /usr/lib/x86_64-linux-gnu/libpython2.7.so
    - Providers:
      - BinDeps.AptGet package python2.7
 - Library "vtkCommon"
    - Satisfied by:
      - System Paths at /usr/bin/../lib/libvtkCommon.so
    - Providers:
      - Simple Build Process
 - Library "ddApp"
    - Providers:
      - Simple Build Process
$ ls deps/usr/lib 
bot2-core                   libddApp.so      liblcm-static.a           librobotlocomotion-lcmtypes.so  python2.7
lcm                         liblcm.so        liblcmtypes_bot2-core.so  libvtkDRCFiltersPythonD.so      robotlocomotion-lcmtypes
libctkPythonConsole.so      liblcm.so.1      libPythonQt.so            libvtkDRCFilters.so
libctkPythonConsole.so.1.0  liblcm.so.1.3.1  libQtPropertyBrowser.so   pkgconfig
```
so `libddApp.so` seems to be there I do not understand why BinDeps is complaining

**Comment by blegat** [comment]
I have written a little C file to see what is the error returned by dlopen and I get
```
could not open file [libddApp.so]: libQVTK.so.5.8: cannot open shared object file: No such file or directory
```
However I have done
```
sudo apt-get install libvtk5-qt4-dev python-vtk
```
Maybe it does not work because I have Ubuntu 16.04

**Comment by rdeits** [comment]
@patmarion have you tried the precompiled binaries on 16.04?

**Comment by patmarion** [comment]
Sorry, I don't expect the binaries to work anywhere except Ubuntu 14, since these binaries have system dependencies.  I can do two things:
- package vtk with director
- use travis-ci's ubuntu 16 image (I assume this exists?) to provide director packages for ubuntu 16

**Comment by patmarion** [comment]
Does your pkg have the ability to select which tarball to download based on ubuntu 14 or 16?

**Comment by rdeits** [comment]
Ah, got it. Yes, it should be easy to select an appropriate version given the Ubuntu version.

**Comment by rdeits** [comment]
@patmarion it actually looks like there isn't a 16.04 travis image: https://github.com/travis-ci/travis-ci/issues/5821 so perhaps bundling vtk is a better option?

**Comment by patmarion** [comment]
Here's a package for ubuntu 16, please let me know how it works!

http://people.csail.mit.edu/patmarion/software/director/releases/director-0.1.0-ubuntu16.tar.gz

**Comment by blegat** [comment]
@patmarion It works great, thanks !

**Comment by rdeits** [comment]
@blegat I'm updating the build.jl script to grab the right binary version on 16.04: #8

**Comment by loicspace** [comment]
I have the same error on Ubtuntu 16.04 but it seems like it is downloading the proper package:

```
INFO: Building DrakeVisualizer
INFO: Attempting to Create directory /home/loic/.julia/v0.5/DrakeVisualizer/deps/downloads
INFO: Downloading file http://people.csail.mit.edu/patmarion/software/director/releases/director-0.1.0-35-g26aa674-ubuntu-16.04.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 9759k  100 9759k    0     0   487k      0  0:00:20  0:00:20 --:--:-- 1218k
INFO: Done downloading file http://people.csail.mit.edu/patmarion/software/director/releases/director-0.1.0-35-g26aa674-ubuntu-16.04.tar.gz
INFO: Attempting to Create directory /home/loic/.julia/v0.5/DrakeVisualizer/deps/usr
=========================================================================================[ ERROR: DrakeVisualizer ]=========================================================================================

LoadError: Provider BinDeps.BuildProcess failed to satisfy dependency ddApp
while loading /home/loic/.julia/v0.5/DrakeVisualizer/deps/build.jl, in expression starting on line 79

```

**Comment by rdeits** [comment]
@loicspace I'm not sure what's going on there, but it's possible that you're running into the same issues that came up on Travis, which were fixed by adjusting the build process slightly. If you have Julia v0.6, you can try the newest build of DrakeVisualizer master by doing:

```julia
Pkg.clone("DrakeVisualizer")
Pkg.build("DrakeVisualizer")
```

which will use the updated build rules. If that works, then we can also back-port that bug fix to the last Julia v0.5-compatible version.
